src/ArrangerConduit.sol

```diff
@@ -0,0 +1,251 @@
+
+        positions[ilk][asset] += amount;
+        totalPositions[asset] += amount;
+    }
+
+    function withdraw(bytes32 ilk, address asset, address destination, uint256 withdrawAmount)
+        external override
+    {
+        // maxWithdraw < pendingWithdrawals < positions < totalWithdrawable < totalPositions
+        require(
+            withdrawAmount <= maxWithdraw[ilk][asset],
+            "Conduit/insufficient-withdrawal"
+        );
+
+        maxWithdraw[ilk][asset]        -= withdrawAmount;
+        pendingWithdrawals[ilk][asset] -= withdrawAmount;
+        positions[ilk][asset]          -= withdrawAmount;
+
+        totalPositions[asset]    -= withdrawAmount;
+        totalWithdrawable[asset] -= withdrawAmount;
+
+        require(
+            ERC20Like(asset).transfer(destination, withdrawAmount),
+            "Conduit/withdraw-transfer-failed"
+        );
+    }
+
+    function requestFunds(bytes32 ilk, address asset, uint256 amount, bytes memory data)
+        external override returns (uint256 fundRequestId)
+    {
+        fundRequestId = fundRequests[asset].length;  // Current length will be the next index
+
+        fundRequests[asset].push(FundRequest({
+            status:          StatusEnum.PENDING,
+            ilk:             ilk,
+            amountRequested: amount,
+            amountFilled:    0
+        }));
+
+        pendingWithdrawals[ilk][asset] += amount;
+
+        require(
+            pendingWithdrawals[ilk][asset] <= positions[ilk][asset],
+            "Conduit/insufficient-position"
+        );
+    }
+
+    function cancelFundRequest(address asset, uint256 fundRequestId) external override {
+        delete fundRequests[asset][fundRequestId];
+    }
+
+    /**********************************************************************************************/
+    /*** Fund Manager Functions                                                                 ***/
+    /**********************************************************************************************/
+
+    function drawFunds(address asset, uint256 amount) external override isFundManager {
+        outstandingPrincipal[asset] += amount;
+
+        require(
+            amount <= ERC20Like(asset).balanceOf(address(this)) - totalWithdrawable[asset],
+            "Conduit/insufficient-available-cash"
+        );
+
+        require(ERC20Like(asset).transfer(fundManager, amount), "Conduit/transfer-failed");
+    }
+
+    function returnFunds(address asset, uint256 returnAmount) external override isFundManager {
+        outstandingPrincipal[asset] -= returnAmount;
+
+        uint256 fundsRemaining = returnAmount;
+
+        uint256 newStartingFundRequestId = startingFundRequestId[asset];
+
+        // For all of the assets fund requests, fill as much as possible
+        // Maintain the order of the fund requests array and update after all fills are complete.
+        for (uint256 i = newStartingFundRequestId; i < fundRequests[asset].length; i++) {
```

Comment:
There's actually no need to maintain the `i` value as you are incrementing via `newStartingFundRequestId`. I would just do a `while(newStartingFundRequestId < length)` with breaks for end conditions.